### ISSUE TITLE
Suggestion: Command to force light recalculations

### ISSUE THREAD

**Issue by esotericist** [maruohon/worldutils#2]
I've had problems off and on over the years with various circumstances coming up where the lighting in a large section of the world was incorrect for a variety of factors (errors in worldgen, mod blocks being removed, etc), and I've never been able to find a tool for correcting that.

Is it possible that functionality could be rolled into this tool? It seems within the design scope.

**Comment by maruohon** [comment]
It would definitely fall into the scope of this mod, yes.

Light recalculations are often pretty much the slowest part of the game, so this command would likely lock up the server pretty good at times, even when trying to schedule things, depending on how large of an area the calculation ends up affecting... unless I were to try and use a custom algorithm for the lighting, which I actually might (have to) do, especially if I want to do this in the "raw Chunk NBT mode" where most of the other operations are also done.

So yes, I'll add this to my TODO list for the mod, but I have no ETA on when I'll have time to work in this mod next.

**Comment by esotericist** [comment]
The performance concerns are understandable, but if I've got a swath of world which is incorrectly lit, I'd be willing to let it run overnight the way I sometimes let pregen tasks run overnight.

I don't actually need this RIGHT NOW, for the record; I went ahead and bit the bullet and generated a new world, but this is a problem I have had repeatedly over the years, and anticipate experiencing again someday.

This -- along with the rest of worldutil -- would be super helpful in those instances.

**Comment by maruohon** [comment]
Yep, and I've actually seen the annoying light issues myself while testing the blockreplace commands and effectively deleting large-ish sections of the world... as that command doesn't do any light updates, it simply changes the block ID arrays etc. in the raw chunk NBT data. So I should probably also add an option to that command to update the light at the same time, instead of having to run the command separately.

And also related to what you mentioned, it is also planned that all/most of the commands could also be run only on selected areas or chunks. But again, if only I had time to work on all the mods and ideas I have written down... :)

**Comment by esotericist** [comment]
Time management, the eternal struggle. I'm sure you'll get there in time.
